chocolatey-diff/Public/Get-ChocolateyPackageDiff.ps1

```diff
@@ -107,6 +107,31 @@
     Expand-Archive -Path $oldFileName -DestinationPath $oldExtractPath -Force
     Expand-Archive -Path $newFileName -DestinationPath $newExtractPath -Force
 
+    foreach ($path in @($oldExtractPath; $newExtractPath)) {
+        # Only because of unit tests
+        if (!(Test-Path $path)) { continue; }
+
+        $metaDestination = Join-Path $path "binary-data.txt"
+        Push-Location -Path $path
+        if (Test-Path $metaDestination) {
+            Remove-Item $metaDestination
+        }
+
+        Get-ChildItem -Path $path -Recurse -File | ? { Test-IsBinary -Path $_.FullName } | % {
+            $item = Get-Item $_.FullName
+            $result = @{
+                path           = Resolve-Path $_.FullName -Relative
+                checksum       = Get-FileHash $_.FullName -Algorithm SHA256 | % Hash
+                fileVersion    = $item.VersionInfo.FileVersion
+                productVersion = $item.VersionInfo.ProductVersion
+            }
+
+            $result
+        } | Out-File -Encoding utf8 -FilePath $metaDestination
+
+        Pop-Location
+    }
+
```

Comment:
I think this should go into it's own function in the Private folder.

Also could you post an example of the file?

Reply:
Sure (using 7zip.install as an example):

```sh
/workspaces/chocodiff> cat /tmp/chocodiff/7zip.install.18.6/binary-data.txt

Name                           Value
----                           -----
path                           ./tools/7zip_x32.exe
fileVersion                    
checksum                       7A2E78C5C2EBC072088B7D8DFF6E00D414E7084C97C2C730CC4C45905668A8F3
productVersion                 
path                           ./tools/7zip_x64.exe
fileVersion                    
checksum                       18D0BF6A9A2BC4DD0779B5FCEFE0AA893F938419D755FE9BC6DBFA50D94BD488
productVersion                 
```

I agree that it should be in its own function, but wanted to get some working and working correctly before I tried to figure out any name for a function.

Reply:
Looks good, I think this could work.